src/MagicScaler/Experimental/TransformFactory.cs

```diff
@@ -21,4 +21,26 @@ public static IPixelSource CreateConversionTransform(IPixelSource source, Guid d
 		ColorProfile? destProf = destProfile.IsEmpty ? null : ColorProfile.Parse(destProfile);
 		return new ConversionTransform(source.AsPixelSource(), PixelFormat.FromGuid(destFormat), sourceProf, destProf);
 	}
+
+	/// <summary>Creates a transform that resizes an <see cref="IPixelSource"/> to the given size, optionally using a specific interpolator.</summary>
+	/// <param name="source">The <see cref="IPixelSource"/> to retrieve pixels from.</param>
+	/// <param name="newWidth">The width of the resulting <see cref="IPixelSource"/>.</param>
+	/// <param name="newHeight">The height of the resulting <see cref="IPixelSource"/>.</param>
+	/// <param name="interpolationSettings">The interpolation settings to use. If <c>null</c>, the default high-quality scaler will be used.</param>
+	/// <returns>An <see cref="IPixelSource"/> that provides the resulting pixel data.</returns>
+	public static IPixelSource CreateScalerTransform(IPixelSource source, int newWidth, int newHeight, InterpolationSettings? interpolationSettings)
+	{
+		if (interpolationSettings is null)
+		{
+			double scaleRatio = Math.Min(
+				source.Width > 0 ? (double)newWidth / source.Width : 0, 
+				source.Height > 0 ? (double)newHeight / source.Height : 0);
+			InterpolationSettings defaultInterpolationSettings = SettingsUtil.GetDefaultInterpolation(scaleRatio);
+			return CreateScalerTransform(source, newWidth, newHeight, defaultInterpolationSettings);
```

Comment:
This recursive call is unnecessary.  The default settings block should just replace the parameter value and fall through.